common/src/app/common/types/container.cljc

```diff
@@ -565,23 +566,36 @@
           (gsh/close-attrs? attr val shape-val 1)
           (gsh/close-attrs? attr val shape-val))
 
+        group-not-equal? (and group (not equal?))
+
         touched?
-        (and group (not equal?) (not (and ignore-geometry is-geometry?)))]
+        (and group-not-equal? (not (and ignore-geometry is-geometry?)))
+
+        content-diff (when (and (= (:type shape) :text) (= attr :content))
+                       (cttc/text-content-diff (:content shape) val))
+
+        groups (cond->> (d/nilv token-groups #{})
+                 group-not-equal?
+                 (cons group)
+
+                 (and group-not-equal? (contains? content-diff :text))
+                 (cons :text-content-text)
 
+                 (and group-not-equal? (contains? content-diff :attribute))
+                 (cons :text-content-attribute))]
     (cond-> shape
       ;; Depending on the origin of the attribute change, we need or not to
       ;; set the "touched" flag for the group the attribute belongs to.
       ;; In some cases we need to ignore touched only if the attribute is
       ;; geometric (position, width or transformation).
       (and in-copy?
-           (or (and group (not equal?)) (seq token-groups))
-           (not ignore?) (not (and ignore-geometry is-geometry?)))
+           (not-empty groups)
+           (not ignore?)
+           (not (and ignore-geometry is-geometry?)))
```

Comment:
This section is very tricky, there are lots of corner cases managed here. Have you checked all combinations?